src/System.Private.DataContractSerialization/src/System/Runtime/Serialization/ReflectionXmlFormatWriter.cs

```diff
@@ -244,54 +273,60 @@ private void ReflectionWriteValue(Type type, object value, bool writeXsiType)
                 }
                 else
                 {
-                    throw new NotImplementedException();
+                    ReflectionInternalSerialize(value, value.GetType().TypeHandle.Equals(memberType.TypeHandle), writeXsiType, memberType);
                 }
             }
             else
             {
                 if (isNullableOfT)
                 {
-                    throw new NotImplementedException();
+                    if (memberValue == null)
+                    {
+                        Type[] genericArgumentTypes = memberType.GetGenericArguments();
+                        if (genericArgumentTypes.Length != 1)
+                        {
+                            throw new InvalidOperationException(string.Format("Cannot serialize type: {0}", memberType.Name));
```

Comment:
Thanks for the suggestions. I agree. I also noticed that there are other exception messages (some have been checked in) we should refactor. I opened https://github.com/dotnet/corefx/issues/9331 for revisiting all the new exceptions before we merging the branch back to master.